e2e/vmss.go

```diff
@@ -198,9 +205,9 @@ func getVmssName(r *mrand.Rand) string {
 	return fmt.Sprintf(vmssNameTemplate, randomLowercaseString(r, 4))
 }
 
-func getBaseVMSSModel(name, location, subscription, mcResourceGroupName, subnetID, sshPublicKey, customData, cseCmd string) armcompute.VirtualMachineScaleSet {
+func getBaseVMSSModel(name, sshPublicKey, customData, cseCmd string, opts *scenarioRunOpts) armcompute.VirtualMachineScaleSet {
 	return armcompute.VirtualMachineScaleSet{
-		Location: to.Ptr(location),
+		Location: to.Ptr(opts.suiteConfig.location),
```

Comment:
Similar question - can opts be nil? Might be worth adding some safeguards to those accesses?

Reply:
again, shouldn't ever be nil since we construct it before running each scenario, though wouldn't be against adding safety around this in the future